src/main/java/io/jenkins/plugins/darktheme/DarkThemeRootAction.java

```diff
@@ -0,0 +1,56 @@
+package io.jenkins.plugins.darktheme;
+
+import hudson.Extension;
+import hudson.model.UnprotectedRootAction;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.Objects;
+import org.apache.commons.io.IOUtils;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.WebMethod;
+
+import static io.jenkins.plugins.darktheme.DarkThemeManagerFactory.THEME_CSS;
+import static io.jenkins.plugins.darktheme.DarkThemeManagerFactory.THEME_URL_NAME;
+import static io.jenkins.plugins.darktheme.DarkThemeSystemManagerFactory.THEME_SYSTEM_CSS;
+import static java.util.Objects.requireNonNull;
+
+@Extension
+public class DarkThemeRootAction implements UnprotectedRootAction {
+
+    @Override
+    public String getIconFileName() {
+        return null;
+    }
+
+    @Override
+    public String getDisplayName() {
+        return null;
+    }
+
+    @Override
+    public String getUrlName() {
+        return THEME_URL_NAME;
+    }
+
+    @WebMethod(name = THEME_CSS)
+    public void doDarkThemeCss(StaplerRequest req, StaplerResponse res) throws IOException {
```

Comment:
This approach will not work well for instances with external resource domains enabled. CC @Wadeck who might advica a better implementation

Reply:
Why not? resource root url is only for userContent and archived artifacts afaik, it doesn't affect actual assets

I've tested it and this works fine with a resource root url if that's the feature you're talking about